### DISCUSSION TITLE
Marker Clustering functionality

### DISCUSSION THREAD

**Discussion by dvbeek** [vue-leaflet/vue-leaflet#223] [Ideas]
It would be nice if the Marker Clustering functionality could be implemented...

https://github.com/Leaflet/Leaflet.markercluster

**Comment by Laxisss** [comment]
i would love that, having many markers makes the app lag so quickly

**Comment by veitbjarsch** [comment]
We could create a plugin for the marker cluster but therefore it is necessary to export the injection keys (and optional but also desirable to export the function folder). 
I created a PR here as a first step to allow external plugin extensions which integrate seamlessly. 
https://github.com/vue-leaflet/vue-leaflet/pull/313

**Reply by 1Luc1** [reply]
Thanks for your work to get Marker Clustering on track again!
 Right now I'm trying to get the [Proof Of Concept](https://codesandbox.io/s/vue-3-leaflet-marker-cluster-poc-fweeu) from [wirk](https://github.com/vue-leaflet/vue-leaflet/issues/111#issuecomment-887794644) to work.

So far i got this: https://pastebin.com/14CmwdUc
No Errors, but doesn't make any marker cluster. You know whats wrong?

**Reply by veitbjarsch** [reply]
With only the file you posted I could only guess. Might be a problem of the versions. I also saw this POC and tried to get it working. But I realized that this POC you posted only works with a specific version subset of the dependencies. It doesn't work with the latest ones. 

I hopefully will be able to release my vue markercluster plugin in the upcoming week (fingers crossed). If you don't have a time critical job you might even can wait for it. But keep in mind that this will only be an alpha and breaking changes will happen high likely in the first minor updates.

**Reply by 1Luc1** [reply]
Thanks for your reply. You are right, posting only the file wasn't the best idea ;) ... to lazy to setup a whole example and put in on github :(.
Since it isn't that time critical for me, I will wait wait for your plugin and see what I can get out of it. Thanks for your efforts 👍 .

**Comment by veitbjarsch** [comment]
@1Luc1 @Laxisss I've created a basic markercluster plugin for vue-leaflet. 
https://github.com/veitbjarsch/vue-leaflet-markercluster

Keep in mind that the plugin is currently an alpha. The basic markercluster works but it still needs a lot of testing and improvement, so every feedback will be welcome. I'll try to address any bugs or suggestions as fast as possible.

**Reply by 1Luc1** [reply]
Thanks for your effort. I already tested it, and for my simple use case it runs very good!

**Reply by veitbjarsch** [reply]
Sounds great. I also ask a friend of mine to test the plugin. Her found some small bugs. But nothing critical. Most of the props should work correctly. 

Once the basic things are stable I'll implement stuff like the capability to handle hundreds/thousands of markers (like the original plugin is capable of).
